pkg/meta/redis.go

```diff
@@ -4240,3 +4247,49 @@ func (m *redisMeta) doAttachDirNode(ctx Context, parent Ino, dstIno Ino, name st
 		return err
 	}, m.inodeKey(parent), m.entryKey(parent)))
 }
+
+// caller makes sure inode is not special inode.
+func (m *redisMeta) touchAtime(ctx Context, ino Ino, cur *Attr) (rerr syscall.Errno) {
+	if (m.conf.AtimeMode != StrictAtime && m.conf.AtimeMode != RelAtime) || m.conf.ReadOnly {
+		return 0
+	}
+
+	var attr *Attr
+	newAttr := &Attr{}
+	if cur != nil {
+		attr = cur
+	} else if m.of.Check(ino, newAttr) {
+		attr = newAttr
+	}
+
+	updated := false
+	defer func() {
+		if rerr == 0 && m.of.IsOpen(ino) && updated {
+			m.of.Update(ino, attr)
+		}
+	}()
+
+	return errno(m.txn(ctx, func(tx *redis.Tx) error {
+		if attr == nil {
```

Comment:
remove the `if` condition, and always get the latest attribute in the transaction

Reply:
Fixed